datafusion/expr/src/window_frame.rs

```diff
@@ -147,6 +147,17 @@ pub enum WindowFrameBound {
     Following(ScalarValue),
 }
 
+impl WindowFrameBound {
+    /// check the frame is UNBOUNDED or not
+    pub fn is_unbounded(&self) -> bool {
+        match self {
+            WindowFrameBound::Preceding(x) => x.eq(&ScalarValue::Null),
+            WindowFrameBound::CurrentRow => false,
+            WindowFrameBound::Following(x) => x.eq(&ScalarValue::Null),
```

Comment:
```suggestion
            WindowFrameBound::Preceding(x) => x.is_null(),
            WindowFrameBound::CurrentRow => false,
            WindowFrameBound::Following(x) => x.is_null(),
```

Reply:
I recommend using `x.is_null()` so that it also catches typed nulls (like `ScalarValue::UInt6(None)`)